quest_test/test_resource_stream.py

```diff
@@ -1,73 +1,377 @@
+import asyncio
 import pytest
 
-from quest import step, Historian
+from quest import Historian
+from quest import step
 from quest.external import state, queue
+from utils import timeout
+
+
+# TODO:
+#  - Make test to assert public resources do not receive updates to a private resource
+#  - Make a test to have a client run on a separate task and just use with: async for...
+
+@step
+async def big_phrase(phrase):
+    return phrase * 3
+
+
+async def default_workflow():
+    async with state('phrase', None, 'woot') as phrase:
+        await phrase.set(await big_phrase(await phrase.get()))
+
+        async with queue('messages', None) as messages:
+            new_message = await messages.get()
+            await phrase.set((await phrase.get()) + new_message)
+
+        await phrase.set('all done')
+
+
+async def mult_identity_workflow():  # A workflow with public and private resources
+    async with state('phrase', None, 'woot') as phrase:
+        await phrase.set(await big_phrase(await phrase.get()))
+
+        async with queue('messages', 'private_identity') as messages:
+            new_message = await messages.get()
+            await phrase.set((await phrase.get()) + new_message)
+
+        await phrase.set('all done')
+
+
+async def mult_private_identity_workflow():  # A workflow with two different private resources
+    async with state('phrase', 'ident1', 'woot') as phrase:
+        await phrase.set(await big_phrase(await phrase.get()))
+
+        async with queue('messages', 'ident2') as messages:
+            new_message = await messages.get()
+            await phrase.set((await phrase.get()) + new_message)
+
+        await phrase.set('all done')
```

Comment:
These workflows are all the same. Perhaps have just one workflow, and you inject the two identities (of which zero to all may be None)?

Reply:
Also, does the workflow need to be so busy? If we can simplify this down to the minimum structure needed for accurate tests, that would be helpful.

Reply:
On the workflow being 'busy' I agree that there is a lot going on. However what should we cut out? Do we need to demonstrate multiple types of Quest resources in a workflow or could we just use states? Do we need to have an @step function for the workflow?